src/server/survey_and_program_data.R

```diff
@@ -143,6 +152,10 @@ surveyAndProgramData <- function(input, output, state, spectrumFilesState) {
     state$programDataValid <- shiny::reactive({
         validateProgramData(state$program_data, spectrumFilesState$countryAndRegionName())
     })
+    
+    state$surveyDataValid <- shiny::reactive({
+      validateSurveyData(state$program_data, spectrumFilesState$countryAndRegionName())
```

Comment:
Oops, this should be `state$survey_data`! Really we could do with some tests for the locking down of navigation (I know we don't have any at the moment.)